frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertScheduleText.tsx

```diff
@@ -0,0 +1,89 @@
+import { useCallback } from "react";
+import _ from "underscore";
+
+import {
+  AM_PM_OPTIONS,
+  HOUR_OPTIONS,
+  getDayOfWeekOptions,
+} from "metabase/lib/date-time";
+import { isNullOrUndefined } from "metabase/lib/types";
+import type { Channel } from "metabase-types/api";
+
+type AlertScheduleTextProps = {
+  schedule: Channel;
+  verbose?: boolean;
+};
+
+export const AlertScheduleText = ({
+  schedule,
+  verbose,
+}: AlertScheduleTextProps) => {
+  const getScheduleText = useCallback(() => {
```

Comment:
imo, I would extract `getScheduleText({ schedule, verbose })` into its own function, so its possible to add tests to this function

also, I think we can do either ts-pattern or early returns here

Reply:
I'll add that in a separate PR so we can focus on that part a little more - I have some code for `ts-pattern` with this function but it's a bit of a mess with the extra null checks and things